src/bios.rs

```diff
@@ -199,9 +133,13 @@ impl Component for Bios {
             anyhow::bail!("Failed to find adoptable system")
         };
 
-        let device = self.get_device()?;
-        let device = device.trim();
-        self.run_grub_install("/", device)?;
+        let target_root = "/";
+        let devices = blockdev::get_backing_devices(&target_root)?
```

Comment:
How about adding a helper function like `blockdev::get_single_backing_device()` that errors out if it finds > 1 (and also errors out if we somehow get zero).

Reply:
Update to use `blockdev::get_single_device()`.